avocado/plugins/xunit.py

```diff
@@ -10,17 +10,109 @@
+        return attrib
+
+    def _escape_cdata(self, cdata):
+        cdata = ''.join(_ if _ in self.PRINTABLE else "\\x%02x" % ord(_)
+                        for _ in str(cdata))
+        return cdata.replace(']]>', ']]>]]&gt;<![CDATA[')
+
+    def _get_attr(self, container, attrib):
+        return self._escape_attr(container.get(attrib, self.UNKNOWN))
+
+    def _create_testcase_element(self, document, state):
+        testcase = document.createElement('testcase')
+        testcase.setAttribute('classname', self._get_attr(state, 'class_name'))
+        testcase.setAttribute('name', self._get_attr(state, 'name'))
+        testcase.setAttribute('time', self._get_attr(state, 'time_elapsed'))
+        return testcase
 
+    def _create_failure_or_error(self, document, test, element_type):
+        element = Element(element_type)
+        element.setAttribute('type', self._get_attr(test, 'fail_class'))
+        element.setAttribute('message', self._get_attr(test, 'fail_reason'))
+        traceback_content = self._escape_cdata(test.get('traceback', self.UNKNOWN))
+        traceback = document.createCDATASection(traceback_content)
+        element.appendChild(traceback)
+        system_out = Element('system-out')
+        system_out_cdata_content = self._escape_cdata(test.get('text_output', self.UNKNOWN))
+        system_out_cdata = document.createCDATASection(system_out_cdata_content)
+        system_out.appendChild(system_out_cdata)
+        element.appendChild(system_out)
+        return element
 
-class XUnit(CLI):
+    def _render(self, result):
+        document = Document()
+        testsuite = document.createElement('testsuite')
+        testsuite.setAttribute('name', 'avocado')
+        testsuite.setAttribute('tests', self._escape_attr(result.tests_total))
+        testsuite.setAttribute('errors', self._escape_attr(result.errors + result.interrupted))
+        testsuite.setAttribute('failures', self._escape_attr(result.failed))
+        testsuite.setAttribute('skipped', self._escape_attr(result.skipped))
+        testsuite.setAttribute('time', self._escape_attr(result.tests_total_time))
+        testsuite.setAttribute('timestamp', self._escape_attr(datetime.datetime.now()))
+        document.appendChild(testsuite)
+        for test in result.tests:
+            testcase = self._create_testcase_element(document, test)
+            status = test.get('status', 'ERROR')
+            if status in ('PASS', 'WARN'):
+                pass
+            elif status == 'SKIP':
+                testcase.appendChild(Element('skipped'))
+            elif status == 'FAIL':
+                element = self._create_failure_or_error(document, test, 'failure')
+                testcase.appendChild(element)
+            else:
+                element = self._create_failure_or_error(document, test, 'error')
+                testcase.appendChild(element)
+            testsuite.appendChild(testcase)
+        return document.toxml(encoding='UTF-8')
```

Comment:
Well one question, why not to use `toprettyxml`? Previously the xml was human readable, this one is ... harder.